hw/ip/otbn/util/shared/dmem_dump.py

```diff
@@ -47,9 +59,11 @@ def parse_actual_dmem(dump: bytes) -> bytes:
     Returns the dmem bytes except integrity info.
     '''
     dmem_bytes = []
+    hexdump = dump[6:]
+    bindump =  bytes.fromhex(hexdump.decode('ascii'))
```

Comment:
Note that this crazy bit of code is because the data as passed to us is not actually `bytes` but hex encoded in ASCII. The actual format is something like
```
dmem:
23abcd...
```
i.e. the first six bytes are literally `b'dmem:\n'` and thus must be skipped, and the remainder is an ASCII string that has to be converted to a number before turning into a bytes object. This feels like there is a mis-alignment somewhere in the APIs but I don't have the requisite knowledge to know where that's happening. This hack allows the singular test case I have run to pass, though, so it is at least points to a way forward?